app/components/steps/rank.js

```diff
@@ -0,0 +1,363 @@
+  const leastCount = () => getRankCount('least')
+
+  const getRankCount = rankName => {
+    return Object.values(rankByParentId).filter(
+      rankedPoint => rankedPoint.category?.toLowerCase() === rankName?.toLowerCase()
+    ).length
+  }
+
+  useEffect(() => {
+    onDone(validAndPercentDone())
+  }, [rankByParentId])
+
+  const validAndPercentDone = () => {
+    let doneCount = 0
+
+    for (const rankedPoint of Object.values(rankByParentId)) {
+      if (rankedPoint.category) doneCount++
+    }
+
+    // Check for difference in expected most/least counts
+    const mostDiscrepancy = mostCount() - targetMost
+    const leastDiscrepancy = leastCount() - targetLeast
+
+    const valid =
+      (mostDiscrepancy == 0 && leastDiscrepancy == 0 && doneCount === pointRankGroupList.length) ||
+      (doneCount === pointRankGroupList.length && targetLeast == 0 && targetMost == 0) // No minimum constraint when there's a single point.
+
+    setRankDiscrepancies({ most: mostDiscrepancy, least: leastDiscrepancy })
+
+    return { valid: valid, percentDone: pointRankGroupList.length ? doneCount / pointRankGroupList.length : 0 } // value should be 0 if not points in list not null
+  }
+
+  // Set the status box error message
+  let errorMsg
+  if (targetLeast > 0 && targetMost > 0) {
+    if (rankDiscrepancies.most > 0 && rankDiscrepancies.least > 0) {
+      errorMsg = `You've rated too many responses as "Most Important" and "Least Important." Please edit.`
+    } else if (rankDiscrepancies.most > 0) {
+      errorMsg = `You've rated too many responses as "Most Important." Please edit.`
+    } else if (rankDiscrepancies.least > 0) {
+      errorMsg = `You've rated too many responses as "Least Important." Please edit.`
+    }
+  }
+
+  return (
+    <div className={cx(classes.rankStep, className)} {...otherProps}>
+      <div className={classes.buttonDiv}>
+        <div className={classes.leftButtons}>
+          <StatusBadge
+            name={'Most Important'}
+            number={`${mostCount()}/${targetMost}`}
+            status={
+              mostCount() == targetMost || (targetLeast == 0 && targetMost == 0)
+                ? 'complete'
+                : mostCount() > targetMost
+                ? 'error'
+                : 'progress'
+            }
+          />
+          <StatusBadge
+            name={'Least Important'}
+            number={`${leastCount()}/${targetLeast}`}
+            status={
+              leastCount() == targetLeast || (targetLeast == 0 && targetMost == 0)
+                ? 'complete'
+                : leastCount() > targetLeast
+                ? 'error'
+                : 'progress'
+            }
+          />
+        </div>
+        <div className={classes.rightButtons}>
+          <ModifierButton
+            className={className.clearButton}
+            title="Clear All"
+            children={'Clear All'}
+            onDone={() => {
+              setRankByParentId({})
```

Comment:
@justin-b-yee I'm not sure if my comment about this got to you before. I don't see it in the messages here.  But I think the solution to the problem is not to setRankByParentId({}) but to iterate through all the items and set the category to ''. (or undefined).
```
setRankByParentId(rankByParentId=>
    Object.values(rankByParentId).reduce(
        (rankByParentId,rank)=>(rankByParentId[rank.parentId]={...rank,category: ''},rankByParentId),
         {}
     )
 )
```
code might be off be a paren or something.